### DISCUSSION TITLE
MCC not possible to run on Raspberry Pi Zero

### DISCUSSION THREAD

**Discussion by Graycot** [ORelio/Minecraft-Console-Client#1988] [General]
Hello! This is my experience from running MCC on a Pi Zero and Pi 3 (not 3+) last year.

I have found that MCC is near impossible to reliably run using Mono on ARM v6 chips (such as the one in the Raspberry Pi Zero and Pi 3). While ARM v6 is somewhat supported by Mono, it appears that not all of the necessary libraries are.

As far as CPU and ram go, the Pi Zero is more than powerful enough to run MCC. I have been running MCC with Mono on a VPS with 512mb ram and 2 slow threads.

The problem: Mono will crash as soon as a the Minecraft account logs onto a server using MCC.

If anyone has a solution this the Armv6 issues on the Pi Zero, I'm all ears!

The new-ish Raspberry Pi Zero 2w has an ARM v7 architecture that should play nicer with Mono. I'll make an update edit to this post when I can get my hands on one for testing.

**Comment by milutinke** [comment]
Hello, this is a quite old thread, I do not know if you will see the reply. 
Now you do not need Mono in order to run the MCC, it does not work anymore, you need to use .NET, which has an ARM build, we had people successfully run the MCC on Raspberry PIs.
As for using/installing it, you either can download an ARM 64 bit binary or build it from the source code.
Take a look at our new documentation website where everything is explained in great detail: https://mccteam.github.io/docs/guide/installation.html

You also can check out this discussion: https://github.com/MCCTeam/Minecraft-Console-Client/discussions/2136

**Reply by Graycot** [reply]
Hello! Thanks for the reply. 

After this semester I hope to tinker with MCC again. Hopefully I'll have the time to finish/fix the step-by-step hosting MCC on a VPS tutorial for beginners. The internal command documentation is nice, I remember the tedious trial and error of figuring out what did what!

**Reply by milutinke** [reply]
> 



> Hello! Thanks for the reply.
> 
> After this semester I hope to tinker with MCC again. Hopefully I'll have the time to finish/fix the step-by-step hosting MCC on a VPS tutorial for beginners. The internal command documentation is nice, I remember the tedious trial and error of figuring out what did what!

I am already working on a section for setting up MCC on a VPS, if you want, you could give feedback/improve upon on it once it's out.
